src/mlpack/core/tree/spill_tree/spill_tree_impl.hpp

```diff
@@ -192,6 +192,87 @@ SpillTree(const SpillTree& other) :
   }
 }
 
+/**
+ * Copy Assignment.
+ */
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename HyperplaneMetricType> class HyperplaneType,
+         template<typename SplitMetricType, typename SplitMatType>
+             class SplitType>
+SpillTree<MetricType, StatisticType, MatType, HyperplaneType, SplitType>&
+SpillTree<MetricType, StatisticType, MatType, HyperplaneType, SplitType>::
+operator=(const SpillTree& other)
+{
+  if (this == &other)
+    return *this;
+
+  // Freeing memory that will not be used anymore.
+  delete dataset;
+  delete pointsIndex;
+  delete left;
+  delete right;
+  delete parent;
```

Comment:
Take a look at the destructor; this is not the right way to handle the memory cleanup.  Specifically `parent` should not be deleted, as that is simply a pointer we hold to the parent node.  The keys are this: any member of a class that is a pointer *might* need its memory freed, but only those pointer members where `new` is called *need* to be deleted.  I know these things are tedious and tricky, so just let me know if I can point you towards any useful references.

Reply:
No, those are just fine, when you call `delete left`, it will call the destructor of `left`, which in turn will call `delete` on *its* children (if it has them). :+1: